pkg/api/authn.go

```diff
@@ -198,84 +373,187 @@ func basicAuthHandler(ctlr *Controller) mux.MiddlewareFunc {
-					}
+			header := request.Header.Get("Authorization")
 
-					userGroups = append(userGroups, ldapgroups...)
+			if (header == "" || header == "Basic Og==") && isMgmtRequested {
+				next.ServeHTTP(response, request)
 
-					ctx := getReqContextWithAuthorization(username, userGroups, request)
-					next.ServeHTTP(response, request.WithContext(ctx)) //nolint:contextcheck
+				return
+			}
 
-					return
-				}
+			action := auth.PullAction
+			if m := request.Method; m != http.MethodGet && m != http.MethodHead {
+				action = auth.PushAction
 			}
 
-			authFail(response, realm, delay)
+			permissions, err := authorizer.Authorize(header, action, name)
+			if err != nil {
+				ctlr.Log.Error().Err(err).Msg("issue parsing Authorization header")
+				response.Header().Set("Content-Type", "application/json")
+				WriteJSON(response, http.StatusInternalServerError, NewErrorList(NewError(UNSUPPORTED)))
+
+				return
+			}
+
+			if !permissions.Allowed {
+				authFail(response, permissions.WWWAuthenticateHeader, 0)
+
+				return
+			}
+
+			amCtx := acCtrlr.getAuthnMdwContext(BEARER, request)
+			next.ServeHTTP(response, request.WithContext(amCtx)) //nolint:contextcheck
 		})
 	}
 }
 
+func noPasswdAuth(config *config.Config) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+			if request.Method == http.MethodOptions {
+				response.WriteHeader(http.StatusNoContent)
+
+				return
+			}
+
+			ctx := getReqContextWithAuthorization("", []string{}, request)
+			// Process request
+			next.ServeHTTP(response, request.WithContext(ctx)) //nolint:contextcheck
+		})
+	}
+}
+
+func (rh *RouteHandler) AuthURLHandler(provider string) http.HandlerFunc {
+	// generate some state (representing the state of the user in your application,
+	// e.g. the page where he was before sending him to login
+	state := func() string {
+		return uuid.New().String()
+	}
+
+	client, ok := rh.c.RelyingParties[provider]
+	if !ok {
+		return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+			response.WriteHeader(http.StatusBadRequest)
+		})
+	}
+
+	return rp.AuthURLHandler(state, client)
+}
+
+func NewRelyingPartyOIDC(config *config.Config, provider string) rp.RelyingParty {
+	if _, ok := config.HTTP.Auth.OpenID.Providers[provider]; !ok {
+		panic(zerr.ErrOpenIDProviderDoesNotExist)
+	}
+
+	var callback string
+
+	switch provider {
+	case "dex":
+		callback = constants.DexCallbackPath
+	default:
+		panic(zerr.ErrOpenIDProviderDoesNotExist)
+	}
+
+	scheme := "http"
+	if config.HTTP.TLS != nil {
+		scheme = "https"
+	}
+
+	clientID := config.HTTP.Auth.OpenID.Providers[provider].ClientID
+	clientSecret := config.HTTP.Auth.OpenID.Providers[provider].ClientSecret
+	scopes := config.HTTP.Auth.OpenID.Providers[provider].Scopes
+	port := config.HTTP.Port
+	issuer := config.HTTP.Auth.OpenID.Providers[provider].Issuer
+	keyPath := ""
```

Comment:
@alexstan12 do we need this? what is its purpose?

Reply:
See https://pkg.go.dev/github.com/caos/oidc/pkg/client/rp#WithJWTProfile . As I understand,  this is a secure mechanism for verifying the authenticity of JWTs issued by an OIDC server.

Reply:
do we need to put it in zot's config?

Reply:
Maybe we should give the admin the possibility to have this check on the issued JWTs. Although, this is not mandatory.